.github/workflows/codeql.yml

```diff
@@ -0,0 +1,81 @@
+name: "CodeQL"
+
+on:
+  push:
+    paths:
+      - '**/*.c'
```

Comment:
why this 'paths' limitations? what about if you change e.g. something in CMake that will affect the compilation...? Is this job too long to be executed for all pushes/PRs?

Reply:
Yes, it's quite long, ca. 20 minutes. And it's running for every push, so I decided to limit this workflow somehow to not run on changes to docs.

As to the cmake, you're right - I can just add CMakeLists.txt to this list. What do you think?